tests/conftest.py

```diff
@@ -65,3 +74,60 @@ def image(dev_image):
 def store_path():
     """Path to store zarr images written from benchmarks"""
     return pathlib.Path("data/output/heart-example.zarr")
+
+
+def expand_min_max(config: dict) -> dict:
+    """Expand min/max keys in config to a list of the full range of values."""
+
+    for key in ["chunk_size", "blosc_clevel", "gzip_level", "zstd_level"]:
+        values = config[key]
+        if "min" in values and "max" in values:
+            config[key] = range(values["min"], values["max"] + 1)
+
+    return config
+
+
+def parse_config_files(config_name: str) -> list[dict]:
+    """Parse json config files. 'all' will parse every config in tests/benchmark_configs (except for dev, which contains
+    a small set of parameters for development runs)."""
+
+    configs_dir = pathlib.Path("tests/benchmarks/benchmark_configs")
+    configs = []
+
+    if config_name == "all":
+        for config_file in configs_dir.iterdir():
+            if config_file.is_file() and config_file.stem != "dev":
+                config = expand_min_max(read_json_file(config_file))
+                configs.append(config)
+    else:
+        config_file = configs_dir / f"{config_name}.json"
+        config = expand_min_max(read_json_file(config_file))
+        configs.append(config)
+
+    return configs
+
+
+def pytest_generate_tests(metafunc):
+    """Parse the config file, and parametrize the given function with these values. pytest_generate_tests is called
+    once per test function, during the collection stage."""
+
+    config_name = metafunc.config.getoption("config")
+    configs = parse_config_files(config_name)
+
+    # keys from the config, that are used as arguments for this function
+    used_config_keys = [key for key in configs[0] if key in metafunc.fixturenames]
+    if len(used_config_keys) == 0:
+        return
```

Comment:
In what case would there be keys in the config that aren't used in the test functions?

Reply:
`pytest_generate_tests` is called in `pytest`'s collection phase, once for every test function it collects.

If we run the benchmarks like below (i.e. without `benchmark-only`):
```
tox -- --dev-image --rounds=1 --warmup-rounds=0
```
It will run the tests under `tests/tests`, as well as the benchmarks under `tests/benchmarks`. When a test is collected e.g. those under `tests/tests/test_write_zarr_array`, they will pass through `pytest_generate_tests`. As they use none of the benchmark parameters, they will trigger the `if len(used_config_keys) == 0` to skip parametrization.